.travis.yml

```diff
@@ -32,6 +32,21 @@ matrix:
       osx_image: xcode7.3
       env: SCALANATIVE_GC=immix
 
+addons:
+  apt:
+    sources:
+    - ubuntu-toolchain-r-test
+    - llvm-toolchain-precise-3.7
+    packages:
+    - clang++-3.7
+    - llvm-3.7
+    - llvm-3.7-dev
+    - llvm-3.7-runtime
+    - llvm-3.7-tools
+    - libgc-dev
+    - libunwind7-dev
+    - make
+
```

Comment:
One of the giant benefits of having unified travis setup script is that we can also share it to simplify 3-rd party builds too. Can we do caching without using `addons: apt: ...` (e.g. by using `before_cache`) ?